dynamic_programming/fibonacci_bottom_up.cpp

```diff
@@ -1,6 +1,34 @@
+/**
+ * @file
+ * @brief Implementation to find n th fibonacci_number
+ * (https://en.wikipedia.org/wiki/Fibonacci_number)
+ *
+ * @details
+ * The Fibonacci sequence is a series of numbers where each number is the
+ * sum of the two preceding ones, starting with 0 and 1.
+ *
+ * ### Algorithm
+ * The idea of the algorithm is to find the ith fibonacci number.
+ * DP state : F(i) = ith fibonacci number
+ * DP state transition :
+ *      F(i) = F(i-1) + F(i-2)
+ * DP base case :
+ *      F(0) = 0
+ *      F(1) = 1
+ * Space reduction :
+ * for i the fibonacci store only i-1 th and i-2 th state.
+ */
+#include <cassert>  /// for assert
 #include <iostream>
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
 using namespace std;
 int fib(int n) {
```

Comment:
Would be awesome if you could document this function according to our guidelines. If you need any help, let us know. 🙂